tools/ctl/ide/assistant/assistant.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2025 Google LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package assistant
+
+import (
+	"context"
+
+	"cloud.google.com/go/vertexai/genai"
+	"github.com/google/oss-rebuild/internal/llm"
+	"github.com/google/oss-rebuild/tools/ctl/ide/chatbox"
+	"github.com/google/oss-rebuild/tools/ctl/localfiles"
+	"github.com/google/oss-rebuild/tools/ctl/rundex"
+	"github.com/pkg/errors"
+)
+
+const (
+	gcpLocation    = "us-central1"
+	geminiIdentity = chatbox.Identity("gemini")
+)
+
+// Session represents one ongoing conversation with the assistant.
+type Session interface {
+	chatbox.ChatBackend
+}
+
+// Assistant provides sessions, debugging different contexts.
+type Assistant interface {
+	Session(context.Context, rundex.Rebuild) (Session, error)
+}
+
+// cmdFunc SHOULD NOT close the out channel. The invoking code may still use the out channel after the cmdFunc has completed.
+// Also, cmdFunc should only return once it no longer needs the `out` channel, as it will be closed soon after.
+type cmdFunc func(ctx context.Context, out chan<- *chatbox.Message) error
+
+type assistant struct {
+	butler  localfiles.Butler
+	project string
+}
+
+var _ Assistant = (*assistant)(nil)
+
+func NewAssistant(butler localfiles.Butler, project string) *assistant {
+	return &assistant{
+		butler:  butler,
+		project: project,
+	}
+}
+
+func (a *assistant) Session(ctx context.Context, attempt rundex.Rebuild) (Session, error) {
+	chat, err := newChat(ctx, a.project)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating llm chat session")
+	}
+	return newSession(a.butler, chat, attempt), nil
+}
+
+func newChat(ctx context.Context, project string) (*llm.Chat, error) {
+	client, err := genai.NewClient(ctx, project, gcpLocation)
```

Comment:
hoist out of here and accept in the constructor. i think it can be shared across sessions.

ew can also then move the GCP location constant out of here and have it be the responsibility of the caller